File: 3163-string-compression-iii/3163-string-compression-iii.py
class Solution:
    def compressedString(self, word: str) -> str:
        if not word:
            return ""
        
        compstr = []
        n = len(word)
        i = 0
        
        while i < n:
            curr = word[i]
            c = 0
            
            while i < n and word[i] == curr and c < 9:
                c += 1
                i += 1
            compstr.append(f"{c}{curr}")
        return ''.join(compstr)

File: 3163-string-compression-iii/3163-string-compression-iii.js
/**
 * @param {string} word
 * @return {string}
 */
var compressedString = function(word) {
    word += "&"
    let r = ""
    let c = 1
    for(let i = 1; i < word.length; i++, c++) {
        if (word[i-1] !== word[i] || c === 9) {
            r += c + word[i-1]
            c = 0
        }
    }
    return r
};